orttraining/orttraining/core/graph/optimizer/lamb_optimizer_builder.cc

```diff
@@ -80,7 +103,16 @@ Status LambOptimizerBuilder::Build(
   // At the end of each Lamb call, the update count may be increased by one.
   const std::string step_tensor_name = "Step";  // per weight optimizer requires a per weight update count
   // Add step as an initializer.
-  new_external_initializers.emplace_back(CreateTensorProto<int64_t>(step_tensor_name, 1));
+  TensorProto step_tensor_proto;
+  const auto step_state_it = shared_optim_state.find(step_tensor_name);
+  if (step_state_it != shared_optim_state.end()) {
+    const auto& init_tensor = step_state_it->second.Get<Tensor>();
+    ORT_THROW_IF_ERROR(IsMatchingTypeAndShape(init_tensor, ONNX_NAMESPACE::TensorProto_DataType_INT64, {1}));
+    step_tensor_proto = utils::TensorToTensorProto(init_tensor, step_tensor_name);
+  } else {
+    step_tensor_proto = CreateTensorProto<int64_t>(step_tensor_name, 1);
+  }
+  new_external_initializers.emplace_back(step_tensor_proto);
```

Comment:
optional: you could use ```std::move``` here so that a copy can be avoided (only if ```step_tensor_proto``` is no longer used here). Optional because maybe the compiler is smart enough to elide the copy here.

Reply:
keeping it unchanged as a step tensor is a single int64